powerauth-push-model/src/main/java/io/getlime/push/model/entity/ListOfUsers.java

```diff
@@ -17,13 +17,20 @@
 package io.getlime.push.model.entity;
 
 import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Request used for adding collection of users to campaign
  *
  * @author Martin Tupy, martin.tupy.work@gmail.com
  */
-
 public class ListOfUsers extends ArrayList<String> {
```

Comment:
I am not sure what to think about this code - why are we extending an ArrayList? It does not seem necessary.

Reply:
I have no strong opinion here - why is extending `ArrayList` bad? The `ListOfUsers` class represents RESTful API entity that is a collection of user identifiers. Having own class for that might have better semantics than just generic `ArrayList<String>`...

Reply:
... plus it is a good way to reference specific type, for example in `ListOfUsersFromCampaignResponse` ([here](https://github.com/lime-company/powerauth-push-server/blob/cfbf5c76ae85f5f0e49b85aa722c968d1bfd8155/powerauth-push-model/src/main/java/io/getlime/push/model/response/ListOfUsersFromCampaignResponse.java)). We also have some responses that are lists themselves, such as `ListOfCampaignsResponse` ([here](https://github.com/lime-company/powerauth-push-server/blob/cfbf5c76ae85f5f0e49b85aa722c968d1bfd8155/powerauth-push-model/src/main/java/io/getlime/push/model/response/ListOfCampaignsResponse.java)).

Reply:
Let's search the internets for answers...

https://www.ibm.com/developerworks/java/library/j-jtp02216/#2.1

https://softwareengineering.stackexchange.com/questions/246277/good-or-bad-practice-to-mask-java-collections-with-meaningful-class-names

Reply:
Ok, so based on the articles, we can probably leave it as is.

- There is often an argument "use composition instead". This is correct for examples like `class School extends ArrayList<Student>`, where students are actually one of many properties of school and using array list seems like an abuse.
- Also, there is an argument about object comparison and lookup. This is not relevant for us, use use these classes as model class for RESTful APIs and we need them to only bind with the underlying JSON.
- Arguments about performance will not be the issue anymore, it is not 2006.

Reply:
Ok